tests/macros/groupdata.test.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * @prettier
+ */
+
+const { itMacro, describeMacro } = require('./utils');
+
+/*
+Different strings in GroupData objects have different sets of
+permitted characters.
+*/
+const permittedCharacters = {
+  group:      /^[\w ()-]+$/,
+  overview:   /^[\w ()-]+$/,
+  interface:  /^[\w.]+$/,
+  property:   /^[\w.]+$/,
+  method:     /^[\w.()]+$/,
+  event:      /^[\w()]+$/,
+  dictionary: /^\w+$/,
+  callback:   /^\w+$/,
+  type:       /^\w+$/,
+  guideTitle: /^[\w .,]+$/,
```

Comment:
Does this allow for colons? Doesn't look like it, but I know we have titles that include them.

Reply:
It does not, but there aren't any guides in here with colons (or the test would fail). I've chosen to be conservative with things like this: rather than permit any characters that it seems we might want to use, I'm permitting the ones we actually do use. If someone adds a guide that contains a colon, they can update the regex then.

Reply:
Makes sense. I do think we need to be sure to note somewhere that we keep track of that the lack of permitting characters in this test isn't meant to suggest a rule about what's allowed in titles and such then.